fixtures/network/manifest/missing_app_uuid.yaml

```diff
@@ -0,0 +1,147 @@
+interactions:
+- request:
+    body: '{"data": {"type": "app_manifest", "attributes": {"manifest_version": "2.0.0",
+      "app_id": "mongodb", "display_on_public_website": true, "tile": {"overview":
+      "README.md#Overview", "configuration": "README.md#Setup", "support": "README.md#Support",
+      "changelog": "CHANGELOG.md", "description": "Track read/write performance, most-used
+      replicas, collection metrics, and more.", "title": "MongoDB", "media": [], "classifier_tags":
+      ["Supported OS::Linux", "Supported OS::macOS", "Supported OS::Windows", "Category::Data
+      Store", "Category::Log Collection", "Category::Autodiscovery"]}, "author": {"support_email":
+      "help@datadoghq.com", "name": "Datadog", "homepage": "https://www.datadoghq.com",
+      "sales_email": "info@datadoghq.com"}, "oauth": {}, "assets": {"integration":
+      {"source_type_name": "MongoDB", "configuration": {"spec": "assets/configuration/spec.yaml"},
+      "events": {"creates_events": true}, "metrics": {"prefix": "mongodb.", "check":
+      "mongodb.connections.available", "metadata_path": "metadata.csv"}, "service_checks":
+      {"metadata_path": "assets/service_checks.json"}, "process_signatures": ["mongod"]},
+      "dashboards": {"mongodb": "assets/dashboards/overview.json"}, "monitors": {"[MongoDB]
+      High incoming connections": "assets/monitors/high_connections.json"}, "saved_views":
+      {"operations_by_type_overview": "assets/saved_views/operations_by_type_overview.json",
+      "queries": "assets/saved_views/queries.json", "queries_by_type_overview": "assets/saved_views/queries_by_type_overview.json",
+      "slow_queries": "assets/saved_views/slow_queries.json", "mongodb_processes":
+      "assets/saved_views/mongodb_processes.json"}, "logs": {"source": "mongodb"}}}}}'
+    headers:
+      accept:
+      - '*/*'
+      accept-encoding:
+      - gzip, deflate
+      connection:
+      - keep-alive
+      content-length:
+      - '1645'
+      content-type:
+      - application/json
+      host:
+      - app.datadoghq.com
+      user-agent:
+      - python-httpx/0.23.1
```

Comment:
We will have to make vcrpy ignore this header otherwise it will fail to match the request when we will use a new version. We can use https://vcrpy.readthedocs.io/en/latest/advanced.html#register-your-own-request-matcher to do so. Something like this: 

```python
def exclude_user_agent_check(request1: Request, request2: Request):
    del headers1['User-Agent']
    del headers2['User-Agent']
    assert headers1 == headers2, f"{headers1} != {headers2}"
```

And in the fixture

```python
        vcr.register_matcher("headers", exclude_user_agent_check)
```

Disclaimer: not tested

Reply:
Since this file now exists, wouldn't it never try the request again? I guess I don't quite understand when we would encounter such an issue

Reply:
AFAIK, when the test will perform an http call, vcrpy will look for this particular request in the cassette, matching the method, headers, url and stuff. If we bump the httpx version in the future, it will likely make the test fail because the header won't match the cassette anymore. You use the record_mode `new_episodes` only to generate the cassette. Once that's done, you have to switch the record_mode to none so vcrpy will only use the cassette and if no requests match the cassette, the test will fail.

Reply:
During implementing this I realized headers are actually not matched on by default https://github.com/kevin1024/vcrpy/blob/42d79b1102a1907376212e0a505851f4f8e81f5d/vcr/config.py#L45

Would you still like me to do this?

Reply:
Ah! My bad then, no need to add that 🙂